src/Features/Core/Portable/ExtractMethod/SelectionResult.cs

```diff
@@ -138,6 +139,61 @@ public bool ShouldPutAsyncModifier()
             return false;
         }
 
+        public bool ShouldCallConfigureAwaitFalse()
+        {
+            var firstToken = GetFirstTokenInSelection();
+            var lastToken = GetLastTokenInSelection();
+
+            var span = TextSpan.FromBounds(firstToken.SpanStart, lastToken.Span.End);
+
+            foreach (var node in SemanticDocument.Root.DescendantNodesAndSelf())
+            {
+                if (!node.Span.OverlapsWith(span))
+                {
+                    continue;
+                }
+
+                if (IsConfigureAwaitFalse(node) && !UnderAnonymousOrLocalMethod(node.GetFirstToken(), firstToken, lastToken))
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        private bool IsConfigureAwaitFalse(SyntaxNode node)
+        {
+            var syntaxFacts = SemanticDocument.Project.LanguageServices.GetService<ISyntaxFactsService>();
+            if (!syntaxFacts.IsInvocationExpression(node))
+            {
+                return false;
+            }
+
+            var invokedExpression = syntaxFacts.GetExpressionOfInvocationExpression(node);
+            if (!syntaxFacts.IsSimpleMemberAccessExpression(invokedExpression))
+            {
+                return false;
+            }
+
+            var name = syntaxFacts.GetNameOfMemberAccessExpression(invokedExpression);
+            var identifier = syntaxFacts.GetIdentifierOfSimpleName(name);
+            if (!syntaxFacts.StringComparer.Equals(identifier.ValueText, nameof(Task.ConfigureAwait)))
+            {
+                return false;
+            }
+
+            var arguments = syntaxFacts.GetArgumentsOfInvocationExpression(node);
+            if (arguments.Count != 1)
+            {
+                return false;
+            }
+
+            var argument = arguments[0];
+            var expression = syntaxFacts.GetExpressionOfArgument(argument);
+            return syntaxFacts.IsFalseLiteralExpression(expression);
```

Comment:
💡 Since this is a refactoring (which only runs at the explicit request of a user), consider using the semantic model to evaluate the argument as a _constant expression_ with a false value. This is optional because it's a rare edge case.

Reply:
I think the only case where this is likely to occur is when there's a global constant which can be set to configure ContinueOnCapturedContext or not.

However I'm not even sure doing this is the right thing in such a case, because you probably want to use the global constant rather than false.

Given that this is extremely rare and would add reasonable complexity to an otherwise simple method, I've left it for now.